src/pathwayvis/components/map/map.component.ts

```diff
@@ -161,13 +161,16 @@ class MapComponentCtrl {
         this._mapOptions.setAddedReactions(reactions);
       }
       if (changes.measured.reactions) {
+        const {reactions} = this._mapOptions.getDataModel();
         // TODO filter out adapter and DM reactions
-        const reactions = changes.measured.reactions.filter((reaction) => {
+        let measuredReactions = changes.measured.reactions.filter((reaction) => {
           return !['adapter', 'DM', 'EX_'].some((str) => {
             return reaction.id.startsWith(str);
           });
         });
-        this._mapOptions.setChangedReactions(reactions);
+        measuredReactions = reactions.filter((reaction) =>
```

Comment:
I need to get the intersection, to get the values of the model to the new bounds, because inside the measuredReactions is the reaction, but with the originals bundles.